diff -Naur original/src/proto_tcp.c patched/src/proto_tcp.c
--- original/src/proto_tcp.c	2015-10-21 05:43:13.000000000 +0800
+++ patched/src/proto_tcp.c	2015-10-31 20:00:54.679653068 +0800
@@ -511,6 +511,9 @@
 	if (global.tune.server_rcvbuf)
                 setsockopt(fd, SOL_SOCKET, SO_RCVBUF, &global.tune.server_rcvbuf, sizeof(global.tune.server_rcvbuf));
 
+	int tos = 4;
+	setsockopt(fd, IPPROTO_IP, IP_TOS, &tos, sizeof(tos));
+
 	if ((connect(fd, (struct sockaddr *)&conn->addr.to, get_addr_len(&conn->addr.to)) == -1) &&
 	    (errno != EINPROGRESS) && (errno != EALREADY) && (errno != EISCONN)) {
 
